10-Reflection/Reflection.Tasks/CommonTasks.cs

```diff
@@ -60,10 +73,28 @@ public static class CommonTasks
         /// <param name="propertyPath">dot-separated property path</param>
         /// <param name="value">assigned value</param>
         public static void SetPropertyValue(this object obj, string propertyPath, object value) {
-            // TODO : Implement SetPropertyValue method
-            throw new NotImplementedException();
-        }
+			// TODO : Implement SetPropertyValue method
+			//throw new NotImplementedException();
+
+			var ass = obj.GetType();
+			var propertyPaths = propertyPath.Split('.');
+
+			if (propertyPaths.Length == 1)
+			{
+				obj.GetType().BaseType.GetProperty(propertyPath).SetValue(obj, value);
```

Comment:
what if you need to go deeper https://allisonmaruska.files.wordpress.com/2015/10/we-need-to-go-deeper.jpg